### ISSUE TITLE
[OpenStack] CCM failing due to empty LoadBalancer provider

### ISSUE THREAD

**Issue by irozzo-1A** [kubermatic/kubermatic#6919]

**Comment by irozzo-1A** [comment]
When creating an OpenStack user cluster with k8s version 1.20 or above the `cloud-controller-manager` fails with the following error:

```
W0422 16:35:13.602968       1 openstack.go:302] failed to read config: Unsupported LoadBalancer Provider:
F0422 16:35:13.603013       1 main.go:107] Cloud provider could not be initialized: could not init cloud provider "openstack": Unsupported LoadBalancer Provider:
goroutine 1 [running]:
k8s.io/klog/v2.stacks(0xc00000e001, 0xc0002e2b40, 0xa3, 0x115)
	/home/ubuntu/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:1026 +0xb9
k8s.io/klog/v2.(*loggingT).output(0x2b74f00, 0xc000000003, 0x0, 0x0, 0xc000658310, 0x2aa983b, 0x7, 0x6b, 0x0)
	/home/ubuntu/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:975 +0x19b
k8s.io/klog/v2.(*loggingT).printf(0x2b74f00, 0xc000000003, 0x0, 0x0, 0x0, 0x0, 0x1d2c0d0, 0x2b, 0xc0006129b0, 0x1, ...)
	/home/ubuntu/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:750 +0x191
k8s.io/klog/v2.Fatalf(...)
	/home/ubuntu/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:1502
main.main.func2(0xc000499b80, 0xc0002cb4c0, 0x0, 0x4)
	/home/ubuntu/go/src/k8s.io/cloud-provider-openstack/cmd/openstack-cloud-controller-manager/main.go:107 +0x2e9
github.com/spf13/cobra.(*Command).execute(0xc000499b80, 0xc00004c0b0, 0x4, 0x4, 0xc000499b80, 0xc00004c0b0)
	/home/ubuntu/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:854 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0xc000499b80, 0xc00053c170, 0xc00053c160, 0x0)
	/home/ubuntu/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/home/ubuntu/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
main.main()
	/home/ubuntu/go/src/k8s.io/cloud-provider-openstack/cmd/openstack-cloud-controller-manager/main.go:141 +0x61d

goroutine 6 [chan receive]:
k8s.io/klog/v2.(*loggingT).flushDaemon(0x2b74f00)
	/home/ubuntu/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:1169 +0x8b
created by k8s.io/klog/v2.init.0
	/home/ubuntu/go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:417 +0xdf

goroutine 76 [select]:
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x1dd0bb8, 0x1f356c0, 0xc00039f3e0, 0x1, 0xc0000940c0)
	/home/ubuntu/go/pkg/mod/k8s.io/apimachinery@v0.20.0/pkg/util/wait/wait.go:167 +0x149
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x1dd0bb8, 0x12a05f200, 0x0, 0xc000502501, 0xc0000940c0)
	/home/ubuntu/go/pkg/mod/k8s.io/apimachinery@v0.20.0/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/home/ubuntu/go/pkg/mod/k8s.io/apimachinery@v0.20.0/pkg/util/wait/wait.go:90
k8s.io/apimachinery/pkg/util/wait.Forever(0x1dd0bb8, 0x12a05f200)
	/home/ubuntu/go/pkg/mod/k8s.io/apimachinery@v0.20.0/pkg/util/wait/wait.go:81 +0x4f
created by k8s.io/component-base/logs.InitLogs
	/home/ubuntu/go/pkg/mod/k8s.io/component-base@v0.20.0/logs/logs.go:58 +0x8a

goroutine 77 [chan receive]:
k8s.io/apimachinery/pkg/watch.(*Broadcaster).loop(0xc0002cb6c0)
	/home/ubuntu/go/pkg/mod/k8s.io/apimachinery@v0.20.0/pkg/watch/mux.go:219 +0x66
created by k8s.io/apimachinery/pkg/watch.NewBroadcaster
	/home/ubuntu/go/pkg/mod/k8s.io/apimachinery@v0.20.0/pkg/watch/mux.go:73 +0xf7

goroutine 78 [chan receive]:
k8s.io/client-go/tools/record.(*eventBroadcasterImpl).StartEventWatcher.func1(0x1f46080, 0xc000202a80, 0xc0005400f0)
	/home/ubuntu/go/pkg/mod/k8s.io/client-go@v0.20.0/tools/record/event.go:301 +0xaa
created by k8s.io/client-go/tools/record.(*eventBroadcasterImpl).StartEventWatcher
	/home/ubuntu/go/pkg/mod/k8s.io/client-go@v0.20.0/tools/record/event.go:299 +0x6e

goroutine 79 [chan receive]:
k8s.io/client-go/tools/record.(*eventBroadcasterImpl).StartEventWatcher.func1(0x1f46080, 0xc000202c30, 0xc000202c00)
	/home/ubuntu/go/pkg/mod/k8s.io/client-go@v0.20.0/tools/record/event.go:301 +0xaa
created by k8s.io/client-go/tools/record.(*eventBroadcasterImpl).StartEventWatcher
	/home/ubuntu/go/pkg/mod/k8s.io/client-go@v0.20.0/tools/record/event.go:299 +0x6e
```

The impacted KKP branches are  `release/v2.17` and `master` at least.

**Comment by irozzo-1A** [comment]
Possibly the problem was introduced here: https://github.com/kubermatic/kubermatic/pull/6885/files#diff-a78ae795f3f480c51a02b4b1cdcbbb91236c658a15484cdc10ac4941330b4497R29 when upgrading the OpenStack CCM. To be checked.

**Comment by irozzo-1A** [comment]
> Possibly the problem was introduced here: https://github.com/kubermatic/kubermatic/pull/6885/files#diff-a78ae795f3f480c51a02b4b1cdcbbb91236c658a15484cdc10ac4941330b4497R29 when upgrading the OpenStack CCM. To be checked.

nope, simply the CCM used for k8s 1.20  (i.e. 1.20.0) was not tolerating empty `lb-provider`, problem fixed in `1.20.2`
